source/codegen/metadata_validation.py

```diff
@@ -0,0 +1,287 @@
+from typing import Any, Dict, Iterable, List, Set
+from schema import Schema, And, Optional, Or, Use
+import common_helpers
+
+
+# Rules that can be suppressed
```

Comment:
note: These are the rules that can be suppressed. If we find cases where other rules need to be suppressed, we can add them as we discover them.